### PR TITLE
Add backend settings endpoint

### PR DESCRIPTION
**Pull request by sbrunner**
## What
- add fixed backend route `/settings` for authenticated users
- add GET `/settings` (returns `{}` for initial config)
- add POST `/settings` to overwrite saved settings JSON
- add configurable payload size limit via `settings.max_payload_size`
- add `user.settings` `JSONB` column in `static` schema
- add alembic migration for the new column
- add functional tests for auth, initial empty settings, overwrite, invalid JSON, and payload too large

Backend part of: https://gitlab.com/geogirafe/gg-viewer/-/work_items/825

See JIRA issue: [GSGIR-93](https://camptocamp.atlassian.net/browse/GSGIR-93).

[GSGIR-93]: https://camptocamp.atlassian.net/browse/GSGIR-93?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: geoportal/tests/functional/test_settings.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+import transaction
+from pyramid import testing
+
+from tests.functional import cleanup_db, create_dummy_request, setup_db
+from tests.functional import setup_common as setup_module  # noqa
+from tests.functional import teardown_common as teardown_module  # noqa
+
+
+class TestSettingsView(TestCase):
+    def setup_method(self, _) -> None:
```
Comment (arnaud-morvan) [inline geoportal/tests/functional/test_settings.py]: Note that other tests create a pytest fixture depending on dbsession and transact, for example https://github.com/camptocamp/c2cgeoportal/blob/master/geoportal/tests/functional/test_themes.py

**Review by arnaud-morvan — APPROVED** [review]
IHMO route `/setting` is ambiguous, I would propose `/user_settings` or `/user/settings` instead.

Otherwise looks good.

File: geoportal/tests/functional/test_settings.py (hunk 1 above)
Comment (arnaud-morvan) [reply geoportal/tests/functional/test_settings.py]: And also for config key in `vars.yaml`:

  `settings` -> `user_settings`.
  `headers/settings` -> `headers/user_settings`
  

And view filename:

  `views/settings.py` -> `views/user_settings.py`

Route names also.

Maybe I forgot something.
